qrenderdoc/Windows/Dialogs/TipsDialog.cpp

```diff
@@ -0,0 +1,323 @@
+/******************************************************************************
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Baldur Karlsson
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ ******************************************************************************/
+
+#include "TipsDialog.h"
+#include "ui_TipsDialog.h"
+
+#include <stdlib.h>
+#include <time.h>
+
+#include <QCloseEvent>
+
+TipsDialog::TipsDialog(CaptureContext &Ctx, QWidget *parent)
+    : m_Ctx(Ctx), QDialog(parent), ui(new Ui::TipsDialog), m_currentTip(0)
+{
+  ui->setupUi(this);
+  setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+  initialize();
+
+  if(m_Ctx.Config.Tips_SeenFirst)
```

Comment:
This logic is kind of backwards from my intuitive understanding of the variable. I read `bool Tips_SeenFirst` as "has the user seen the first tip" - so initially `false`, and then `true` once we've shown it. So we check `!SeenFirst` and show the 0th tip, otherwise we show a random tip.

It is entirely equivalent to define it this way around since it just changes the sense of the `if`, but if you want it this way around I think you should rename the variable to something more understandable otherwise it could lead to people being confused.